### PR TITLE
Code snippet for autolog and some retrofitting to include model.fit() where necessary

### PR DESCRIPTION
**Pull request by dmatrix**
## What changes are proposed in this pull request?

Code snippet for auto logging and retrofit in some methods to fit the model
Components 

- [x] `area/docs`: MLflow documentation pages

- [x] `rn/none`: No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section

### CODE REVIEW

File: mlflow/tracking/fluent.py [hunk 1]
```diff
@@ -1115,6 +1115,49 @@ def autolog(
...
+
+        run_id: 98a1ff588c68471ea44e8121b95cd54f
+        params: {'copy_X': 'True', 'normalize': 'False', 'fit_intercept': 'True',
+                 'n_jobs': 'None'}
+        metrics: {'training_score': 1.0, 'training_rmse': 4.440892098500626e-16,
+                  'training_r2_score': 1.0,
+                  'training_mae': 2.220446049250313e-16, 'training_mse': 1.9721522630525295e-31}
+        tags: {'estimator_class': 'sklearn.linear_model._base.LinearRegression',
+               'estimator_name': 'LinearRegression'}
+        artifacts: ['model/MLmodel', 'model/conda.yaml', 'model/model.pkl']
```
Comment (harupy) [inline mlflow/tracking/fluent.py:1160]: Can we prettify the output?

```python
params: {'copy_X': 'True',
         'normalize': 'False',
         'fit_intercept': 'True',
         'n_jobs': 'None'}

metrics: {'training_score': 1.0,
          'training_rmse': 4.440892098500626e-16,
          'training_r2_score': 1.0,
          'training_mae': 2.220446049250313e-16,
          'training_mse': 1.9721522630525295e-31}
```

Comment (dmatrix) [reply mlflow/tracking/fluent.py:1160]: @harupy You mean use pprint or print dict items per line?

Comment (harupy) [reply mlflow/tracking/fluent.py]: I don't think we need to use `pprint`. We can just edit the output to make it look pretty.

Comment (dmatrix) [reply mlflow/tracking/fluent.py]: Haha. That’s easy (and cheating). :-) I’ll revert back & massage the output.

**Review by harupy — APPROVED** [review]
lgtm

**Pull request merged**